lib/shelley/src/Cardano/Wallet/Shelley/Network.hs

```diff
@@ -777,6 +781,31 @@ debounce action = do
         unless (Just cur == prev) $ action cur
         atomically $ putTMVar mvar (Just cur)
 
+-- | Convenience function to measure the time of a LSQ query,
+-- and trace the result.
+--
+-- Such that we can get logs like:
+-- >>> Query getAccountBalance took 51.664463s
+--
+-- Failures that stop the >>= continuation will cause the corresponding
+-- measuremens /not/ to be logged.
+timeQryAndLog
+    :: MonadIO m
+    => String
+    -- ^ Label to identify the query
+    -> Tracer m (NetworkLayerLog sc)
+    -- ^ Tracer to which the measurement will be logged
+    -> m a
+    -- ^ The action that submits the query.
+    -> m a
+timeQryAndLog label tr act = do
```

Comment:
What about using the `bracketTracer` from the logging like in: https://github.com/input-output-hk/cardano-wallet/blob/master/lib/core/src/Cardano/DB/Sqlite.hs#L222-L223

The main benefits is that you still get some log even if the action doesn't finish and we don't have to compute time on top of the logger which already comes with timestamps.

Reply:
Interesting, but then we'd need some state machine to write `Trace BracketLog -> Trace NominalDiffTime`. That seems _more_ complicated.

I think logging the `NominalDiffTime` is much better than requiring us to look and compare timestamps later.

We could add `MonadCatch` to `timeQryAndLog`, but:
- We don't expect the query `send`s to throw in `IO`
- When it does (I think socket disconnects), do we really care about the time?
- I think `(MonadCatch m, MonadIO m)` wouldn't catch `IO` exceptions if  `m ~ ExceptT`, which could be surprising (we could make `timeQryAndLog` work with a concrete `IO`, but 🤷‍♂️ )